src/components/notification-list-item.component.js

```diff
@@ -70,41 +76,36 @@ export class NotificationListItem extends Component {
 
   render() {
     const { notification, iconAction, navigationAction } = this.props;
-
-    const TitleComponent = this.getComponentType();
+    const tag = this.getComponentType();
     const iconName = this.getIconName(notification.subject.type);
+    const titleComponentProps =
```

Comment:
Hey @LeoCp welcome!
Can you please move this into a class method and add proper tests to it? I don't feel comfortable with all this logic under `render() { ... }`